dts/bindings/gpio/ambiq,gpio.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023 Antmicro <www.antmicro.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: Ambiq GPIO
+
+compatible: "ambiq,gpio"
+
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+  reg:
+    required: true
+    description: GPIO pin offset
```

Comment:
This feels like a slight misuse of the `reg` property, which has well-defined meaning.

Reply:
This was originally changed because of https://github.com/zephyrproject-rtos/zephyr/pull/61157#issuecomment-1677318814 and https://github.com/zephyrproject-rtos/zephyr/pull/61157#issuecomment-1680156614 Can you take another look at the whole conversation?